includes/embeds/class-amp-youtube-embed-handler.php

```diff
@@ -151,10 +245,59 @@ public function render( $args, $url ) {
 				'width'       => $this->args['width'],
 				'height'      => $this->args['height'],
 				'placeholder' => '',
+				'start'       => 0,
 			]
 		);
 
-		if ( empty( $args['video_id'] ) ) {
+		$attributes = array_merge(
+			[ 'data-videoid' => $args['video_id'] ],
+			wp_array_slice_assoc( $args, [ 'layout', 'width', 'height' ] )
+		);
+		if ( ! empty( $args['title'] ) ) {
+			$attributes['title'] = $args['title'];
+		}
+
+		$query_vars  = [];
+		$query_param = wp_parse_url( $url, PHP_URL_QUERY );
+		wp_parse_str( $query_param, $query_vars );
+		$query_vars = ( ! empty( $query_vars ) && is_array( $query_vars ) ) ? $query_vars : [];
+
+		$excluded_param = [ 'autoplay', 'loop', 'start', 'v', 'vi', 'w', 'h' ];
+
+		foreach ( $query_vars as $key => $value ) {
+
+			if ( in_array( $key, $excluded_param, true ) ) {
+				continue;
+			}
+
+			$attributes[ "data-param-$key" ] = sanitize_text_field( $value );
+		}
+
+		foreach ( [ 'autoplay', 'loop' ] as $param ) {
+			if ( isset( $query_vars[ $param ] ) ) {
+				$attributes[ $param ] = $query_vars[ $param ];
+			}
+		}
+
+		if ( ! empty( $args['start'] ) && 0 < (int) $args['start'] ) {
+			$attributes['data-param-start'] = (int) $args['start'];
+		}
```

Comment:
These could be combined into one for loop:

```suggestion
		$excluded_params = [ 'v', 'vi', 'w', 'h' ];

		foreach ( $query_vars as $key => $value ) {
			if ( in_array( $key, $excluded_params, true ) ) {
				continue;
			}

			if ( in_array( $key, [ 'autoplay', 'loop' ], true ) ) {
				$attributes[ $key ] = $value;
				continue;
			}

			if ( 'start' === $key && 0 < (int) $args['start'] ) {
				$attributes['data-param-start'] = (int) $args['start'];
				continue;
			}

			$attributes[ "data-param-$key" ] = esc_attr( $value );
		}
```